tools/standalone/Standalone/Utils.hs

```diff
@@ -0,0 +1,291 @@
+someED25519Pair =
+    ( PubBS $ getByteString
+        "368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca"
+    , PrivBS $ getByteString
+        "251a920c403ae8c8f65f59142316af3c82b631fba46ddea92ee8c95035bd2898"
+    , "368820f80c324bbc7c2b0610688a7da43e39f91d118732671cd9c7500ff43cca"
+    , ED25519
+    )
+
+getByteString :: ByteString -> ByteString
+getByteString = fst . B16.decode
+
+formatB16PubKey :: SomeKeyPair -> Text
+formatB16PubKey = toB16Text . formatPublicKey
+
+onlyCoinTransferMemPoolAccess :: ChainId -> Int -> MemPoolAccess
+onlyCoinTransferMemPoolAccess cid blocksize = MemPoolAccess
+    { mpaGetBlock = \_ _ _ _ -> getTestBlock (chainIdToText cid) blocksize
+    , mpaSetLastHeader = const $ return ()
+    , mpaProcessFork = const $ return ()
+    }
+
+defaultMemPoolAccess :: ChainId -> Int -> MemPoolAccess
+defaultMemPoolAccess cid blocksize  = MemPoolAccess
+    { mpaGetBlock = \_preblockcheck height _hash _prevBlock ->
+        makeBlock height cid blocksize ("(+ 1 2)", Nothing)
+    , mpaSetLastHeader = const $ return ()
+    , mpaProcessFork = const $ return ()
+    }
+  where
+    makeBlock
+        :: BlockHeight
+        -> ChainId
+        -> Int
+        -> (Text, Maybe Value)
+        -> IO (Vector.Vector ChainwebTransaction)
+    makeBlock height cidd n = Vector.replicateM n . go
+        where
+          go (c, d) = do
+              let dd = mergeObjects (toList d)
+                  pm = def
+                    & set pmSender "sender00"
+                    & set pmGasLimit 100
+                    & set pmGasPrice 0.1
+                    & set pmChainId (PactChain.ChainId (chainIdToText cidd))
+                  msg = Exec (ExecMsg c dd)
+                  -- TODO: This might need to be something more fleshed out.
+                  nonce = T.pack $ show height
+              ks <- testKeyPairs
+              cmd <- mkCommand ks pm nonce msg
+              case verifyCommand cmd of
+                ProcSucc t -> return $ fmap (k t) (SB.toShort <$> cmd)
+                ProcFail e -> throwM $ userError e
+
+          k t bs = PayloadWithText bs (_cmdPayload t)
+
+mkPactExecutionService' :: TQueue RequestMsg -> PactExecutionService
+mkPactExecutionService' q = emptyPactExecutionService
+  { _pactValidateBlock = \h pd -> do
+      mv <- validateBlock h pd q
+      r <- takeMVar mv
+      case r of
+          (Right !pdo) -> return pdo
+          Left e -> throwM e
+  , _pactNewBlock = \m h -> do
+      mv <- newBlock m h q
+      r <- takeMVar mv
+      case r of
+          (Right !pdo) -> return pdo
+          Left e -> throwM e
+  }
+
+data StopState
+  = BlockStopCondition BlockStopState
+  | TimeLength Int -- this is in terms of microseconds
+  | Forever
+  deriving (Eq, Show, Generic, ToJSON, FromJSON)
+
+data BlockStopState
+  = Height BlockHeight
+  | Weight BlockWeight
+  deriving (Eq, Show, Generic, ToJSON, FromJSON)
+
+stopAtBlockWeight :: BlockWeight -> CutDb cas -> IO ()
```

Comment:
The implementation actually considers the weight of the whole `cut`, so `stopAtCutWeight` seems to a  more appropriate name.

```suggestion
stopAtCutWeight :: BlockWeight -> CutDb cas -> IO ()
```

Reply:
Changed.